iocore/net/SSLNetVConnection.cc

```diff
@@ -1694,6 +1694,12 @@ SSLNetVConnection::callHooks(TSEvent eventId)
   Debug("ssl", "sslHandshakeHookState=%d eventID=%d", this->sslHandshakeHookState, eventId);
 
   // Move state if it is appropriate
+  if (eventId == TS_EVENT_VCONN_CLOSE) {
+    // Regardless of state, if the connection is closing, then transition to
+    // the DONE state. This will trigger us to call the appropriate cleanup
+    // routines.
+    this->sslHandshakeHookState = HANDSHAKE_HOOKS_DONE;
+  }
```

Comment:
Looks reasonable. We may want to put the `switch` statement below into `else` clause to ensure `sslHandshakeHookState` won't be accidentally overwritten in the `switch` statement.

Reply:
Good idea.